### PR TITLE
Rework python tests.

### PR DESCRIPTION
**Pull request by vsafonkin**
Rework for using Pester test framework.

Example of tests with report (without building part): [link](https://buildcanary.visualstudio.com/CanaryBuilds/_build/results?buildId=825442&view=results)

### CODE REVIEW

File: tests/sources/Python.Tests.ps1 [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    return $LASTEXITCODE
+}
+
+Describe "Python toolcache tests" {
+
+    It "Python version" {
+        & python --version
+        & which python
+        $whichPython = & which python
+        $expectedPath = "$ToolsDirectory/Python"
```
Comment (maxim-lobanov) [inline tests/sources/Python.Tests.ps1]: I would suggest to rewrite it a bit so each call should be checked via Pester:
```
$pythonLocation = (Get-Command "python").Path
$pythonLocation | Should -Not -BeNullOrEmpty
InvokePythonCode "python --version" | Should -Be 0
```

File: tests/sources/Python.Tests.ps1 [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+function InvokePythonCode {
+    Param (
+      [String] [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()]
+      $Command
+    )
+  
+    $null = Invoke-Expression -Command $Command
+    return $LASTEXITCODE
+}
```
Comment (maxim-lobanov) [inline tests/sources/Python.Tests.ps1]: I think it should be renamed because actually this function is not related to Python. I would suggest to rename it to something like `GetCommandExitCode` and ideally move to helper

File: tests/sources/Python.Tests.ps1 [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    It "Check Tkinter" {
+        if ($Platform -notmatch "windows") {
+            InvokePythonCode -Command "python ./check_tkinter.py" | Should -Be 0
+        }
+    }
+
+    It "Validate Pyinstaller" {
+        # Pyinstaller 3.5 does not support Python 3.8.0. Check issue https://github.com/pyinstaller/pyinstaller/issues/4311
+        if ($Version -lt "3.8.0") {
+            Invoke-Expression "pip install pyinstaller"
```
Comment (maxim-lobanov) [inline tests/sources/Python.Tests.ps1]: Why do not we use `InvokePythonCode "pip install pyinstaller" | Should -Be 0`?

File: tests/sources/Python.Tests.ps1 [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+            InvokePythonCode -Command "pyinstaller --onefile ./main.py" | Should -Be 0
+            Invoke-Expression "./dist/main"
+            $LASTEXITCODE | Should -Be 0
```
Comment (maxim-lobanov) [inline tests/sources/Python.Tests.ps1]: The same as above

Comment (maxim-lobanov) [reply tests/sources/Python.Tests.ps1]: And below too :)

File: tests/sources/Python.Tests.ps1 [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+        }
+    }
+
+    It "Pip psutil installation" {
+        if ($Platform -notmatch "windows") {
```
Comment (maxim-lobanov) [inline tests/sources/Python.Tests.ps1]: Does it make sense to move this check to helpers  to simplify code `if (IsNixPlatform) {` ?

File: tests/Run-Tests.ps1 [hunk 6]
```diff
@@ -0,0 +1,14 @@
+param (
+    [Version] [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()]
+    $Version,
+    [String] [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()]
+    $Platform,
+    [String] [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()]
+    $ToolsDirectory
+)
+
+Install-Module Pester -Force -Scope CurrentUser
```
Comment (maxim-lobanov) [inline tests/Run-Tests.ps1]: Is there any machines where Pester is not available?

File: tests/Run-Tests.ps1 [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+
+Set-Location 'tests/sources'
+Import-Module Pester
+Invoke-Pester -Script @{Path='./Python.Tests.ps1'; Parameters=@{Version=$Version; Platform=$Platform; ToolsDirectory=$ToolsDirectory}} -OutputFile 'test_results.xml' -OutputFormat NUnitXml
```
Comment (maxim-lobanov) [inline tests/Run-Tests.ps1]: Do we really need the whole script for these 3 lines?

Comment (vmapetr) [reply tests/Run-Tests.ps1]: Depending on `Install-Module Pester -Force -Scope CurrentUser`. If we installing pester, I suggest, to leave all execution in separate script, in case if pester is installed everywhere, we can simply invoke it from template.

File: tests/Run-Tests.ps1 [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+    $Version,
+    [String] [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()]
+    $Platform,
+    [String] [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()]
+    $ToolsDirectory
+)
+
+Install-Module Pester -Force -Scope CurrentUser
+
+Set-Location 'tests/sources'
```
Comment (vmapetr) [inline tests/Run-Tests.ps1]: I think, it can be moved to `workingDirectory:` argument of AzDO `Run tests` task.
